.github/workflows/checks.yml

```diff
@@ -36,6 +36,10 @@ jobs:
     name: "Unit Tests"
     runs-on: ubuntu-latest
 
+    env:
+      TF_APPEND_USER_AGENT: INTEGRATION-TEST
+      OPENTOFU_USER_AGENT: OPENTOFU
```

Comment:
I think if we're appending, we dont need to override the OPENTOFU_USER_AGENT too

Reply:
Test cases would still need to be changed because TF_APPEND_USER_AGENT is also being tested here 

Also, does unsetting the environment variable not defeat the purpose of having tests for them?

<img width="873" alt="image" src="https://github.com/opentofu/opentofu/assets/72958686/7ea9a286-0200-465a-a7d5-bc6b18620d84">